source/block-reconciler.ts

```diff
@@ -1,95 +1,99 @@
+	if (isFirstBlock(blockHistory)) return await addNewHeadBlock(blockHistory, newBlock, onBlockAdded, blockRetention)
 
 	if (isOlderThanOldestBlock(blockHistory, newBlock)) {
-		blockHistory = await rollback(blockHistory, onBlockRemoved);
-		return await addNewHeadBlock(blockHistory, newBlock, onBlockAdded, blockRetention);
+		blockHistory = await rollback(blockHistory, onBlockRemoved)
+		return await addNewHeadBlock(blockHistory, newBlock, onBlockAdded, blockRetention)
 	}
 
-	if (isAlreadyInHistory(blockHistory, newBlock))
-		return blockHistory;
+	if (isAlreadyInHistory(blockHistory, newBlock)) return blockHistory
 
-	if (isNewHeadBlock(blockHistory, newBlock))
-		return await addNewHeadBlock(blockHistory, newBlock, onBlockAdded, blockRetention);
+	if (isNewHeadBlock(blockHistory, newBlock)) return await addNewHeadBlock(blockHistory, newBlock, onBlockAdded, blockRetention)
 
 	if (parentHashIsInHistory(blockHistory, newBlock)) {
 		while (blockHistory.last().hash !== newBlock.parentHash) {
-			blockHistory = await removeHeadBlock(blockHistory, onBlockRemoved);
+			blockHistory = await removeHeadBlock(blockHistory, onBlockRemoved)
 		}
-		return await addNewHeadBlock(blockHistory, newBlock, onBlockAdded, blockRetention);
+		return await addNewHeadBlock(blockHistory, newBlock, onBlockAdded, blockRetention)
 	}
 
-	return await backfill(getBlockByHash, blockHistory, newBlock, onBlockAdded, onBlockRemoved, blockRetention);
+	return await backfill(getBlockByHash, blockHistory, newBlock, onBlockAdded, onBlockRemoved, blockRetention)
 }
 
 const rollback = async <TBlock extends Block>(blockHistory: BlockHistory<TBlock>, onBlockRemoved: (block: TBlock) => Promise<void>): Promise<BlockHistory<TBlock>> => {
 	while (!blockHistory.isEmpty()) {
-		// CONSIDER: if this throws an exception, removals may have been announced that are actually still in history since throwing will result in no history update. we can't catch errors here because there isn't a clear way to recover from them, the failure may be a downstream system telling us that the block removal isn't possible because they are in a bad state. we could try re-announcing the successfully added blocks, but there would still be a problem with the failed block (should it be re-announced?) and the addition announcements may also fail
-		blockHistory = await removeHeadBlock(blockHistory, onBlockRemoved);
+		// CONSIDER: if this throws an exception, removals may have been announced that are actually still in
+		// history since throwing will result in no history update. we can't catch errors here because there
+		// isn't a clear way to recover from them, the failure may be a downstream system telling us that the
+		// block removal isn't possible because they are in a bad state. we could try re-announcing the
+		// successfully added blocks, but there would still be a problem with the failed block (should it be
+		// re-announced?) and the addition announcements may also fail
```

Comment:
Prose text should be one line (soft-wrapped).  Same for other comments throughout.